Lib/uuid.py

```diff
@@ -388,6 +392,48 @@ def _find_mac(command, args, hw_identifiers, get_index):
         pass
     return first_local_mac or None
 
+def _find_mac_netstat(command, args, hw_identifiers, get_index):
```

Comment:
At least in this PR's current form, it seems that the values given for the two new parameters, `hw_identifiers` and `get_index`, are always the same (`b'Address'` and `lambda x: x` respectively; I'm considering `lambda x: x + 0` equivalent to `lambda x: x` since the inputs are always integers.)

Now that this is working, please try to "unwind" some of the changes made while getting to this point, to see if things work properly without them.

Start with replacing `lambda x: x + 0` with `lambda x: x`, which I would be surprised if had any effect.

Then try moving this logic back into `_netstat_getnode()`, since apparently the additional parameterization is no longer needed.

Reply:
testing is working on AIX. Pushing for CI

Reply:
Are you still working on this? My original comment here is still relevant.

Reply:
Misunderstood your request.

I'll try testing it without the lambda statement. Understanding it was hard - but it seems on Linux the position it identifies needs to be increased by one to find the actual value in the following lines. With AIX the positions are equal.

Supposing the function can work without the lambda statement I would still want to consider leaving it in so that "lambda" is used consistently throughout the file.

During my initial debugging/learning I used lambda x: x, but this failed until I used lambda x: x * 1, or lambda x: x + 0.

In any case, I am not trying to be fancy - merely consistent with previous code.

Reply:
I've read this more thoroughly and now understand that this is modified copy/paste of the existing `_find_mac()` function. This should be avoided if reasonably possible.

If the only significant change is the mac address parsing due to different formatting, perhaps we could just add a mac address parser as a parameter to `_find_mac()`? E.g.:

```python
def _find_mac(command, args, hw_identifiers, get_index,
              parse_mac=lambda word: int(word.replace(b':', b''), 16)):
...
```

Reply:
Finally - found where the email came from. ;) I guess this is part of the mentor process - find the comments.

This is "double" as I did reply to the email - but I'll add my response here for clarity.
-
Yes, the routine is based on _find_mac(). But _find_mac is designed for data that has the keyword and the value on the same line. Programs such as netstat have a different layout. And yes, AIX also has different formatting - using the character '.' to divide the fields and also a variable length (1.2.3.4.5.6 is a valid format on AIX (and would be evaluated as 0x010203040506)

The (linux (what I found)) ifconfig commands,as well as the mock-data have the keyword and
the value on a single line while netstat (on both AIX and macos) have the keyword on the "header" line
and in that "position" the keyword on a later line. When there are fewer words, even if it looks as if it is
that column, and/or is not formatted properly (not six 'words' separated

Snippet from macos:
`Name  Mtu   Network       Address            Ipkts Ierrs    Opkts Oerrs  Coll
lo0   16384 <Link#1>                         56907     0    56907     0     0
lo0   16384 127           localhost          56907     -    56907     -     -
...
lo0   16384 127.94.0.1/32 127.94.0.1         56907     -    56907     -     -
                        224.0.0.251
                        all-systems.mcast.net
gif0* 1280  <Link#2>                             0     0        0     0     0
stf0* 1280  <Link#3>                             0     0        0     0     0
XHC20 0     <Link#4>                             0     0        0     0     0
en2   1500  <Link#5>    1a:00:01:0d:82:10        0     0        0     0     0
en3   1500  <Link#6>    1a:00:01:0d:82:11        0     0        0     0     0
en0   1500  <Link#7>    38:c9:86:2b:38:f5  3816938     0  3025627     0     0
`
If called with this data the value 0x1a00010d8210 would be used (IF it _is_universal(0x1a00010d8210) OR not _is_universal(0x38c9862b38f5)).

Snippet from AIX:
`Name  Mtu   Network     Address           Ipkts Ierrs    Opkts Oerrs  Coll
en0   1500  link#2      fa.d1.8c.f7.62.4 129294604     0 57078842     0     0
`
Here the value returned is 0xfad18cf76204 (not 0xfad18cf7624)
by a character).

Further, I have always tried to say away from something that could be
construed as a feature change. From my perspective it is hard enough to
get a straight forward change considered. A feature change seems an
impossibility - e.g., immediate concerns about a potential backwards
compatibility issue. I leave changes of that caliber to others who have
been involved much longer and deeper than myself.

Hope this answers your question. In any case, I await your response before considering to "resolve conversation"

Sincerely,
Michael

Reply:
It was not a "copy" of the _find_mac routine. Instead, I moved the "_find_mac" logic that was in the _netstat_getnode() function - to make all of the "program" related functions more similiar (calling a find_mac_{inline|nextlines) based on whether the keyword (aka hw_indentifier) that used to determine the index that is passed to the "lanbda" function get_index().

Long sentence: in the "inline" version, the keyword and the index are in the same line; in "nextlines" the keyword only appears once in a "header" line.

Once I figured out that distinction - the code was clearer to me - and I hope now, to others as well.